drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -227,14 +228,38 @@ struct bcm2835_isp_fmt *find_format_by_fourcc(unsigned int fourcc,
 	return NULL;
 }
 
+static const
+struct bcm2835_isp_fmt *find_format_by_fourcc_and_cs(unsigned int fourcc,
+						     enum v4l2_colorspace cs,
+						     struct bcm2835_isp_node *node)
+{
+	const struct bcm2835_isp_fmt *fmt;
+	unsigned int i;
+
+	for (i = 0; i < node->num_supported_fmts; i++) {
+		fmt = node->supported_fmts[i];
+		if (fmt->fourcc == fourcc) {
+			if (cs == V4L2_COLORSPACE_DEFAULT ||
+			    (fmt->colorspace_mask & V4L2_COLORSPACE_MASK(cs)))
+				return fmt;
+			else
+				return NULL;
```

Comment:
Assuming I'm understanding this correctly, if you call try_fmt with ```V4L2_PIX_FMT_Y10P``` but colour space ```V4L2_COLORSPACE_MASK_SMPTE170M``` you get back the default format instead, which is most likely going to be ```V4L2_PIX_FMT_YUV420```. That seems wrong.

```find_format``` can return the correct ```struct bcm2835_isp_fmt```, but then try_fmt ought to validate that the colourspace is valid for this format, and correct it if not (I think you've already got that bit in place, but this is going to mess it up).

Reply:
Yes, that makes more sense. I'll fix that up, rebase everything, and re-push!